toolkit/docs/building/building.md

```diff
@@ -636,6 +635,7 @@ To reproduce an ISO build, run the same make invocation as before, but set:
 | UNATTENDED_INSTALLER          |                                                                                                        | Create unattended ISO installer if set. Overrides all other installer options.
 | PACKAGE_BUILD_LIST            |                                                                                                        | Additional packages to build. The package will be skipped if the build system thinks it is already up-to-date.
 | PACKAGE_REBUILD_LIST          |                                                                                                        | Always rebuild this package, even if it is up-to-date. Base package name, will match all virtual packages produced as well.
+| SRPM_PACK_LIST                |                                                                                                        | List of spec basenames to build into SRPMs. If empty, all specs under `$(SPECS_DIR)` will be packed.
```

Comment:
Looks good...just want to double check on how well this interplays with the DELTA_BUILD or HYDRATED_BUILD option.  (Ok to say they are mutually exclusive, just make sure to add some kind of check and fail if someone tries to combine them)

Reply:
**Quick summary of how SRPM_PACK_LIST will interplay technically:**

If running with `DELTA_BUILD=y`, packages will use upstream repos as a cache to resolve nodes in the package graph when possible. By using `SRPM_PACK_LIST` to limit which SRPMs are built, the set of RPMs that need to be downloaded/marked as cached by DELTA_BUILD=y is restricted to what is required by that subset of SRPMs.

If running with `HYDRATED_BUILD=y`, each package is built in its own little microcosm using cached packages, replacing all Run nodes with Prebuilt nodes. By using `SRPM_PACK_LIST`, we reduce the set of SRPMs that need to be processed, and thus we also reduce the number of Run nodes that need to be marked as cached.

**Quick summary of how SRPM_PACK_LIST will interplay semantically with how we use them in the pipelines:**

I don't think `DELTA_BUILD=y` and `SRPM_PACK_LIST` make sense to use together. The point of a delta build is to detect packages that have changed and are unavailable upstream, then rebuild those packages. If we already know what has changed, then why are we using `DELTA_BUILD=y`? It's a bit redundant in purpose.

I also don't think that `HYDRATED_BUILD=y` makes sense to use with `SRPM_PACK_LIST`. If we want to specify that we're only running certain tests, we already have the option to use `PACKAGE_(RE)BUILD_LIST` to do so.

**Now what?**

So, technically, we're fine to use SRPM_PACK_LIST in conjunction with those two flags. In terms of using the delta/hydrated builds for their intended purpose, however, it doesn't make sense to use the flags in this way. Despite this, I don't think we should have these flags be mutually exclusive with the packing list. 

I think it's important to remember the use case for `SRPM_PACK_LIST`. First and foremost, this is a way to get faster local validation builds for our devs by (a) cutting down the number of SRPMs to build and (b) cutting down the size of the package graph that's considered. We don't really have much use for `SRPM_PACK_LIST` in our official pipelines, since we have blazing fast connections to PMC and our source blobstore there. If a dev has a valid reason to do a delta/hydrated build on their local machine, and they also want to save time by using `SRPM_PACK_LIST`, then I don't think we should prevent that. I don't foresee harm/confusion arising from these flags being used together.